e2e/Makefile

```diff
@@ -21,49 +17,35 @@ start-kind: ## Start kind cluster
 
 test: e2e-image ## Run e2e tests against current kube context
 	$(MAKE) -C ../ docker.build \
-		IMAGE_REGISTRY=local/external-secrets \
```

Comment:
Don't we want to use local images for normal e2e tests? If people right now have access to a external secret provider and want to run e2e tests locally they can, without having to push to a registry. Like in this comment: https://github.com/external-secrets/external-secrets/pull/430#issuecomment-975735470

Reply:
The `test` target just builds the image locally and sideloads it into the kind cluster. It doesn't get pushed into a registry - it just uses the same name for consistency and ease of use.

Only the `test.managed` target pushes it into the registry.